crates/project_panel/src/project_panel.rs

```diff
@@ -82,6 +82,7 @@ pub struct ProjectPanel {
     show_scrollbar: bool,
     scrollbar_drag_thumb_offset: Rc<Cell<Option<f32>>>,
     hide_scrollbar_task: Option<Task<()>>,
+    diagnostic_error_colors: HashMap<PathBuf, Color>,
```

Comment:
if we do want to support showing warnings in the project panel, this should we a HashMap. If we only want to show errors we can make this a HashSet and not store the colors.